lnrpc/gen_protos.sh

```diff
@@ -1,34 +1,56 @@
-#!/bin/sh
-
-echo "Generating root gRPC server protos"
-
-# Generate the protos.
-protoc -I/usr/local/include -I. \
-       -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-       --go_out=plugins=grpc,paths=source_relative:. \
-       rpc.proto
-
-# Generate the REST reverse proxy.
-protoc -I/usr/local/include -I. \
-       -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-       --grpc-gateway_out=logtostderr=true,paths=source_relative:. \
-       rpc.proto
-
-# Finally, generate the swagger file which describes the REST API in detail.
-protoc -I/usr/local/include -I. \
-       -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-       --swagger_out=logtostderr=true:. \
-       rpc.proto
-
-# For each of the sub-servers, we then generate their protos, but a restricted
-# set as they don't yet require REST proxies, or swagger docs.
-for file in **/*.proto
-do
-    DIRECTORY=$(dirname ${file})
-    echo "Generating protos from ${file}, into ${DIRECTORY}"
-
-    protoc -I/usr/local/include -I. \
-           -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-           --go_out=plugins=grpc,paths=source_relative:. \
-           ${file}
-done
+#!usr/bin/env sh
+
+build-protoc-gen-go() {
+
+    echo "Install protoc-gen-go"
+    mkdir -p bin
+    export GOBIN=$PWD/bin
+    go install github.com/golang/protobuf/protoc-gen-go \
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway \
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+    
+}
+
+generate() {
+
+    export GOOGLEAPIS=$(find / -path "*third_party/googleapis")
```

Comment:
Nope. You should defintely **never** do a find starting in root, specially in an aux script.

The previous version of this used `$GOPATH` so you can use that or if using the current dir as source for the code, just use `$PWD`.

Reply:
I do this because the script can't find the link from the right version, with `find` I grab the correct absolute path.